coreax/metrics.py

```diff
@@ -95,86 +91,104 @@ def __init__(self, kernel: coreax.kernel.Kernel, precision_threshold: float = 1e
+        num_comparison_points = len(comparison_points)
+        comparison_weights = jnp.atleast_1d(comparison_data.weights)
+        if jnp.allclose(
+            comparison_weights,
+            jnp.broadcast_to(1 / num_comparison_points, (num_comparison_points,)),
+        ):
+            comparison_weights = None
+
+        if comparison_weights is None:
+            if block_size is None or block_size > max(
+                num_reference_points, num_comparison_points
+            ):
+                return self.maximum_mean_discrepancy(
+                    reference_points, comparison_points
+                )
+            return self.maximum_mean_discrepancy_block(
+                reference_points, comparison_points, block_size
+            )
 
         if (
             block_size is None
-            or weights_x is None
-            or block_size > max(num_points_x, num_points_y)
+            or reference_weights is None
+            or block_size > max(num_reference_points, num_comparison_points)
         ):
-            return self.weighted_maximum_mean_discrepancy(x, y, weights_y)
+            return self.weighted_maximum_mean_discrepancy(
+                reference_points, comparison_points, comparison_weights
+            )
 
         return self.weighted_maximum_mean_discrepancy_block(
-            x, y, weights_x, weights_y, block_size
+            reference_points,
+            comparison_points,
+            reference_weights,
+            comparison_weights,
+            block_size,
         )
 
-    def maximum_mean_discrepancy(self, x: ArrayLike, y: ArrayLike) -> Array:
+    def maximum_mean_discrepancy(
+        self, reference_points: ArrayLike, comparison_points: ArrayLike
+    ) -> Array:
         r"""
         Calculate standard, unweighted MMD.
 
-        For a dataset of ``n`` points in ``d`` dimensions, :math:`x`, and another
-        dataset :math:`y` of ``m`` points in ``d`` dimensions, the maximum mean
-        discrepancy is given by:
+        For a dataset of ``n`` points in ``d`` dimensions, :math:`\mathcal_{D}_1`, and
+        another dataset :math:`\mathcal_{D}_2` of ``m`` points in ``d`` dimensions, the
+        maximum mean discrepancy is given by:
 
         .. math::
 
-            \text{MMD}^2(x,y) = \mathbb{E}(k(x,x)) + \mathbb{E}(k(y,y))
-            - 2\mathbb{E}(k(x,y))
+            \text{MMD}^2(\mathcal_{D}_1,\mathcal_{D}_2) = \mathbb{E}(k(\mathcal_{D}_1,
+            \mathcal_{D}_1)) + \mathbb{E}(k(\mathcal_{D}_2,\mathcal_{D}_2))
+            - 2\mathbb{E}(k(\mathcal_{D}_1,\mathcal_{D}_2))
 
         where :math:`k` is the selected kernel.
 
-        :param x: The original :math:`n \times d` data
-        :param y: An :math:`m \times d` array defining a representation of ``x``, for
-            example a coreset
+        :param reference_points: The original :math:`n \times d` data
+        :param comparison_points: An :math:`m \times d` array to compare to
+            ``reference_points``
```

Comment:
We've lost the extra context provided by the original docstring - is this intentional? Same question for the other MMD methods

Reply:
Yes this was intentional as there is no need for comparison_points to be a coreset of reference_points. A valid use case of something like Kernel Herding would be to build coresets of two separate datasets and then compute MMD between these two coresets to do a faster test of dataset similarity; here the docstring wouldn't make sense. Moreover, the functions are symmetric in their _points arguments anyway i.e. even if we had a coreset, the coreset could go first and nothing would be different.

Reply:
Fair enough! I'll point out that `weighted_maximum_mean_discrepancy` is not symmetric, but that should be obvious from the signature. On further inspection, the "...for example a coreset" comment also appears in the docstrings of `maximum_mean_discrepancy_block` and `weighted_maximum_mean_discrepancy_block` - do we want to remove these as well? If so, I think it could be sensible to include a quick line in the class docstring or in the docstring for `compute` explaining possible usages (coreset vs. whole dataset, coreset vs. coreset etc)

Reply:
Yeah the change that Tom C is working on should change a lot of this so probably worth leaving until thats been pushed